src/ucp/api/ucp.h

```diff
@@ -747,7 +750,9 @@ typedef struct ucp_worker_params {
      * This value is optional.
      * If it's not set (along with its corresponding bit in the field_mask -
      * UCP_WORKER_PARAM_FIELD_THREAD_MODE), the UCS_THREAD_MODE_SINGLE mode
-     * will be used.
+     * will be used. Note that actual thread mode may be different from mode
+     * passed to @ref ucp_worker_create. To get actual thread mode use
+     * @ref ucp_worker_query.
      */
```

Comment:
@hoopoepg @yosefe 

Text suggestion : 

> The parameter thread_mode suggests the thread safety mode with which worker and the associated resources should be created with. This is an optional parameter. The default value is UCS_THREAD_MODE_SINGLE and it is used when the value of the parameter is not set. When this parameter along with its corresponding bit in the field_mask - UCP_WORKER_PARAM_FIELD_THREAD_MODE is set, the @ref ucp_worker_create attempts to create worker with this thread mode. The thread mode with which worker is created can differ from the suggested mode. The actual thread mode of the worker should be obtained using the query interface  @ref ucp_worker_query. 


Also, should consider changing thread_mode -> suggested_thread_mode and thread_mode (in ucp_worker_attr_t) -> provided_thread_mode. I'm not sure, if it works for all contexts.

Reply:
@yosefe do you like this wording?

Reply:
"The parameter thread_mode suggests the thread safety mode with which worker and the associated resources should be created with. This is an optional parameter."  - the word "with" appears twice. other than this it's ok
@manjugv we can't change field names because of API compatibility

Reply:
updated